driver-core/src/main/com/mongodb/internal/time/Timeout.java

```diff
@@ -47,32 +53,41 @@ default Timeout orEarlier(@Nullable final Timeout other) {
     }
 
     /**
-     * @see TimePoint#isInfinite()
+     * @see TimePoint#isInfiniteLocal()
      */
-    boolean isInfinite();
-
-    /**
-     * @see TimePoint#hasExpired()
-     */
-    boolean hasExpired();
+    // TODO-CSOT refactor: make static, move to tests
+    @Deprecated
+    default boolean isInfinite() {
+        return run(NANOSECONDS, () -> {
+            return true;
+        }, (ns) -> {
+            return false;
+        }, () -> {
+            return false;
+        });
+    }
 
     /**
-     * @see TimePoint#remaining
+     * @see TimePoint#hasExpiredLocal()
      */
-    long remaining(TimeUnit unit);
+    // TODO-CSOT refactor: make static, move to tests
+    @Deprecated
+    default boolean hasExpired() {
```

Comment:
Suggest keeping this but renaming it so its usage is clear that its a point in time check. eg: `standaloneCheckHasExpired()` (also advocate adding `standaloneCheckHasExpired(TimeUnit)`). 

I think its easier than having to use the `run` for this usecase.

Reply:
Replied to similar topic [here](https://github.com/mongodb/mongo-java-driver/pull/1349#discussion_r1534736064)